daemon/rpc_groups.go

```diff
@@ -13,12 +13,30 @@ import (
 // Groups provides endpoint and autocompletion.
 func (r *RPC) Groups(ctx context.Context, in *pb.GroupsRequest) (*pb.Payload, error) {
 	var cfg config.Config
+	if r.cm == nil || r.dm == nil {
```

Comment:
I don't think this check is necessary actually. At least we are not doing such checks elsewhere. Of course there's a theoretical possibility of a null pointer panic, but in practice this struct has a "constructor" so the actual possibility of them being nil is close to zero unless you are doing something unwise deliberately.
My opinion in general is that checking every time if every used interface is nil is just cumbersome, and in practice at least I never got burned by not checking them :smile:

Reply:
yes, this only happened while writing the tests. I'll remove the check in this case.